pkg/rtc/room.go

```diff
@@ -148,6 +153,30 @@ func NewRoom(
 		r.protoRoom.CreationTime = time.Now().Unix()
 	}
 
+	if agentClient != nil {
+		go func() {
+			res := r.agentClient.CheckEnabled(context.Background(), &rpc.CheckEnabledRequest{})
+			if res.PublisherEnabled {
+				r.lock.Lock()
+				r.publisherAgentsEnabled.Store(true)
+				// if there are already published tracks, start the agents
+				r.hasPublished.Range(func(k, v interface{}) bool {
```

Comment:
without locks, we could fire this multiple times. I think we need to use a lock here to synchronize setting publisherAgentsEnabled, and actually launching the agent.